openmetadata-service/src/main/java/org/openmetadata/service/search/indexes/MlModelIndex.java

```diff
@@ -32,9 +32,7 @@ public Map<String, Object> buildSearchIndexDocInternal(Map<String, Object> doc)
     ParseTags parseTags = new ParseTags(Entity.getEntityTags(Entity.MLMODEL, mlModel));
     Map<String, Object> commonAttributes = getCommonAttributesMap(mlModel, Entity.MLMODEL);
     doc.putAll(commonAttributes);
-    doc.put(
```

Comment:
@harshach,  wanted to understand why we are changing the logic for displayName. In cases where displayName is missing, we usually use name as a fallback.

The reason for this is that we will now be using displayName.keyword in most places. If an entity exists but doesn’t have a displayName, it will not appear in the aggregation results. (when field passed in aggregate api = displayName.keyword)

We can move this common logic to SearchIndexApp.getCommonAttributesMap to avoid repeating it in all index files.

<img width="757" alt="image" src="https://github.com/user-attachments/assets/46ffa4eb-7a5a-4d72-8470-c2f17f947fc9">

Reply:
our search query counts as the search string appears twice and scores it twice

Reply:
@harshach I think we are avoiding the double scoring here - screenshot added below 

, so it should be okay, if we have  
doc.put("displayName",
        mlModel.getDisplayName() != null ? mlModel.getDisplayName() : mlModel.getName());  present
        
        
<img width="805" alt="image" src="https://github.com/user-attachments/assets/fa19b038-d1d6-46c7-805d-532339a8f64a">

Reply:
the above is not going work. We need to make a tie breaker between displayName and name.
for now added displayName back, we will default to displayName